simulariumio/springsalad/springsalad_converter.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+
+from ..trajectory_converter import TrajectoryConverter
+from ..data_objects import TrajectoryData, AgentData, MetaData, UnitData
+from .springsalad_data import SpringsaladData
+
+###############################################################################
+
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class SpringsaladConverter(TrajectoryConverter):
+    def __init__(self, input_data: SpringsaladData):
+        """
+        This object reads simulation trajectory outputs
+        from SpringSaLaD (https://vcell.org/ssalad)
+        and plot data and writes them in the JSON format used
+        by the Simularium viewer
+
+        Parameters
+        ----------
+        input_data : SpringsaladData
+            An object containing info for reading
+            SpringSaLaD simulation trajectory outputs and plot data
+        """
+        self._data = self._read(input_data)
+
+    def _parse_springsalad_data(
+        self, springsalad_data: List[str], input_data: SpringsaladData
+    ) -> Tuple[AgentData, np.ndarray]:
+        """
+        Parse SpringSaLaD SIM_VIEW txt file to get the total steps
+        and maximum agents per timestep
+        """
+        columns_list = [
+            "time",
+            "unique_id",
+            "type",
+            "positionX",
+            "positionY",
+            "positionZ",
+            "radius",
+        ]
+        traj = pd.DataFrame([], columns=columns_list)
+        timestamp = 0.0
+        box_size = np.zeros(3)
+        for line in springsalad_data:
+            cols = line.split()
+            if "xsize" in line:
+                box_size[0] = input_data.scale_factor * 2 * float(cols[1])
+            if "ysize" in line:
+                box_size[1] = input_data.scale_factor * 2 * float(cols[1])
+            if "z_outside" in line:
+                box_size[2] += input_data.scale_factor * 2 * float(cols[1])
+            if "z_inside" in line:
+                box_size[2] += input_data.scale_factor * 2 * float(cols[1])
```

Comment:
i don't know what this code is doing but it looks fishy

Reply:
Also why is everything being doubled?

Reply:
SpringSaLaD output files have bounding volume parameters like:

`xsize	50.0
ysize	50.0
z_outside	10.0
z_inside	90.0`

and by trial and error I found that they're the box "radius" rather than dimensions, that's why they are multiplied by 2. And then the Z radius is z_inside + z_outside. It seems weird to me, but it's how the engine does it?

Reply:
could they mean "inside the cell" and "outside the cell"?

Reply:
yes I'm almost sure that's what they mean